### ISSUE TITLE
Removing the structures with large distortions

### ISSUE THREAD

**Issue by darjaved** [prisms-center/CASMcode#357]
Should i remove the structures with very large distortions, as CE is not able to predict those correctly. few are the DFT ground state structures also and CE is not predicting them, what should i do? any suggestions? how should i remove them?

**Comment by xivh** [comment]
Yes, you can use the `atomic_deformation` and `lattice_deformation` properties to find these structures. The cutoff values will depend on your system, so you can find those by selecting a few distorted structures and seeing what the deformation scores are.

**Comment by darjaved** [comment]
How should i decide the cutoff to remove any structure?

**Comment by xivh** [comment]
You could try fitting iteratively to see how the fit performs, or find the score for a distortion that you know about. You can query `struc_score` to map onto other potential prims and use that as a cutoff. For example, if you are doing a BCC cluster expansion, you can find the deformation scores to HCP and FCC prims and use that for the cutoff values.

**Comment by darjaved** [comment]
{
    "atomic_deformation_cost" : 0.166758947604,
    "best_score" : -43.362576690000,
    "energy" : -43.362576690000,
    "final_path" : "/data/javeedd/CE/Na_Sn_data/casm_0.5/G56/properties.calc.json",
    "has_all_required_properties" : true,
    "has_any_required_properties" : true,
    "import_additional_files" : false,
    "import_properties" : true,
    "import_structure_file" : false,
    "initial_path" : "/data/javeedd/CE/Na_Sn_data/casm_0.5/G56/properties.calc.json",
    "is_new_best" : true,
    "is_new_config" : true,
    "lattice_deformation_cost" : 0.085088893640,
    "preexisting_data" : false,
    "preexisting_files" : false,
    "properties_origin" : "/data/javeedd/CE/Na_Sn_data/casm_0.5/G56/properties.calc.json",
    "score" : -43.362576690000,
    "selected" : false,
    "to_configname" : "SCEL8_8_1_1_0_4_0/4"
  },
  {
    "atomic_deformation_cost" : 0.320702395235,
    "best_score" : -43.677587780000,
    "energy" : -43.658807650000,
    "final_path" : "/data/javeedd/CE/Na_Sn_data/casm_0.5/G57/properties.calc.json",
    "has_all_required_properties" : true,
    "has_any_required_properties" : true,
    "import_additional_files" : false,
    "import_properties" : true,
    "import_structure_file" : false,
    "initial_path" : "/data/javeedd/CE/Na_Sn_data/casm_0.5/G57/properties.calc.json",
    "is_new_best" : false,
    "is_new_config" : true,
    "lattice_deformation_cost" : 0.007317252087,
    "preexisting_data" : false,
    "preexisting_files" : false,
    "properties_origin" : "/data/javeedd/CE/Na_Sn_data/casm_0.5/G57/properties.calc.json",
    "score" : -43.658807650000,
    "selected" : false,
    "to_configname" : "SCEL8_2_4_1_3_1_0/34"
    
    As can be seen in the above structures the cost values. if wonder is this much atomic deformation cost ok?
    cost directly determines the distortion?

**Comment by xivh** [comment]
That score is quite large. If you look at the relaxed structure, is it similar to the starting structure? Also, is the structure near your convex hull?

**Comment by darjaved** [comment]
No, the relaxed structures looks highly distorted. Mostly the structures in my case have this much deformation costs, but CE could predict their energies except some structures, i have RMS value of 11 meV/atom. Structures near the hull are highly distorted ones having the atomic deformation scores more than 0.3.

**Comment by xivh** [comment]
Is it possible that there is another parent crystal structure for that region? For example, you can have relaxations going to FCC, BCC, HCP - you would need a separate project for each parent.

Also, do you have vacancies? If so, you may be finding highly distorted structures such as [Na15Sn4](https://next-gen.materialsproject.org/materials/mp-30794).

**Comment by darjaved** [comment]
What do you mean by another parent crystal? Sn has two crystal structures i.e., BCT and DC, so i took the BCT one and constructed my CE based on that, but there are large distortions almost at every Na composition. but CE could predict the energies with an RMS of 10 meV/atom. there is no clear trend between the predicted energy and the distortions. 

No, i don't have vacancies.

**Comment by darjaved** [comment]
There are some intermetallics for this system. Moreover, since CE could predict the energies of distorted structures within a reasonable error, what should i do? By including the configurations with the very large distortion and CE predicting them with very less error, aren't we predicting the energy of the configuration incorrectly?

**Comment by xivh** [comment]
> What do you mean by another parent crystal? Sn has two crystal structures i.e., BCT and DC, so i took the BCT one and constructed my CE based on that

We usually make one cluster expansion for each parent structure, such as one for FCC and one for BCC, and then filter the relaxed structures so that they match the parent structure.

> Moreover, since CE could predict the energies of distorted structures within a reasonable error, what should i do?

If you are satisfied with the error, then I think it is okay, but just be aware that with deformation scores that large you could be mixing different parent crystal structures together.
